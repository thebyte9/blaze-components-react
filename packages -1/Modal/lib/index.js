"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
exports.__esModule = true;
var utils_1 = __importDefault(require("@blaze-react/utils"));
var react_1 = __importStar(require("react"));
var ModalFooter_1 = __importDefault(require("./ModalFooter"));
var ModalHeader_1 = __importDefault(require("./ModalHeader"));
var Modal = function (_a) {
    var children = _a.children, isSimple = _a.isSimple, isAlert = _a.isAlert, title = _a.title, isUpload = _a.isUpload, actions = _a.actions, overlay = _a.overlay, classNames = _a.utils.classNames, _b = _a.onClose, onClose = _b === void 0 ? function () { return ({}); } : _b;
    var sections = ["header", "content", "footer"];
    var closeModal = function () { return onClose(); };
    var modalClassNames = classNames("modal modal--show", {
        "modal--alert": isAlert,
        "modal--simple": isSimple,
        "modal--upload": isUpload
    });
    var _c = sections.map(function (alertType) {
        var _a;
        return classNames("modal__" + alertType, (_a = {},
            _a["modal__" + alertType + "--alert"] = isAlert,
            _a["modal__" + alertType + "--simple"] = isSimple,
            _a["modal__" + alertType + "--upload"] = isUpload,
            _a));
    }), modalHeaderClassNames = _c[0], modalContentClassNames = _c[1], modalFooterClassNames = _c[2];
    return (react_1["default"].createElement(react_1.Fragment, null,
        overlay && react_1["default"].createElement("div", { className: "overlay", onClick: closeModal }),
        react_1["default"].createElement("div", { className: modalClassNames },
            react_1["default"].createElement("div", { className: modalHeaderClassNames }, !isAlert && react_1["default"].createElement(ModalHeader_1["default"], { title: title, closeModal: closeModal })),
            react_1["default"].createElement("div", { className: modalContentClassNames }, children),
            react_1["default"].createElement(ModalFooter_1["default"], { isAlert: isAlert, footerClassNames: modalFooterClassNames, closeModal: closeModal, actions: actions }))));
};
Modal.defaultProps = {
    actions: [],
    buttonModifiers: ["outline"],
    children: "No content",
    isAlert: false,
    isSimple: false,
    isUpload: false,
    overlay: true,
    title: ""
};
exports["default"] = utils_1["default"](Modal);
