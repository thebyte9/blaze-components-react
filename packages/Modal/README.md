## Description

A modal dialog is a dialog that appears at the top of the main content.
Use a modal for dialog boxes, forms, confirmation messages or other content that can be accessed.

## Usage

- Simple

```js
const actions = [
  {
    callback: () => ({}),
    textButton: "Action 1"
  },
  {
    callback: () => ({}),
    textButton: "Action 2"
  }
];

<Modal title="Simple Modal" actions={actions} IsSimple>
  <p>lorem ipsum dolor...</p>
</Modal>;
```

- Alert

```js
const alertActions = [
  {
    callback: () => ({}),
    modifiers: "alert small",
    textButton: "delete"
  }
];

<Modal actions={alertActions} buttonModifiers="rounded alert" isAlert>
  <p>Delete item?</p>
</Modal>;
```

- Scrollable

```js
<Modal
  title="Scrollable Modal"
  buttonModifiers="outline dark rounded"
  actions={actions}
>
  <p>
    Content here that may need to be scrolled, can be text/forms/etc. Default
    modal with scrollable content:
  </p>
  <ol>
    <li>Lorem</li>
    <li>Ipsum</li>
    <li>Dolor</li>
  </ol>
  ...etc
</Modal>
```

## API

##### Modal can receive a number of `props` as follow:

| NAME         |   TYPE   | DEFAULT  |
| :----------- | :------: | :------: |
| title        |  String  |  empty   |
| actions      |  Array   |    []    |
| isAlert      | Boolean  |  false   |
| isSimple     | Boolean  |  false   |
| isUpload     | Boolean  |  false   |
| isFullScreen | Boolean  |  false   |
| onClose      | Function | () => {} |
