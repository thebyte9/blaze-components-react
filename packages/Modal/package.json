{
  "name": "@blaze-react/modal",
  "version": "0.4.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "",
  "main": "./lib/index.js",
  "scripts": {
    "build": "../../node_modules/.bin/tsc --jsx react --esModuleInterop ./src/*.tsx ./src/**/*.tsx  --outDir lib ",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "../../node_modules/.bin/tslint -c ../../tslint.json src/**/*.tsx",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "pre-publish": "npm run build && npm run lint && npm run test"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@blaze-react/button": "^0.4.0",
    "@blaze-react/utils": "^0.4.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.5"
  },
  "peerDependencies": {
    "prop-types": "^15.7.2",
    "react": "^16.9.0",
    "react-dom": "^16.9.0"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      "@babel/preset-react"
    ]
  },
  "gitHead": "3638834d2211cc16d965ecafbf5ab2eeddceca12"
}
