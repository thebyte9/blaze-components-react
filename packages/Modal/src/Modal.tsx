import buildClassNames from '../../Utils/src/buildClassNames';
import React, { useEffect } from "react";
import { ESCAPE_KEY_CODE } from "./constants";
import ModalFooter from "./ModalFooter";
import ModalHeader from "./ModalHeader";

interface IActions {
  textButton: string;
  callback: () => void;
  modifiers?: string[];
}

interface IModalProps {
  title?: string;
  actions: IActions[];
  isSimple?: boolean;
  isUpload?: boolean;
  isFullScreen?: boolean;
  isAlert?: boolean;
  overlay?: boolean;
  children?: any;
  onClose?: () => void;
  showFooter?: boolean;
  className?: string;
}

const Modal: React.SFC<IModalProps> = ({
  children,
  isSimple,
  isAlert,
  isFullScreen,
  title,
  isUpload,
  actions,
  overlay,
  className,
  showFooter,
  onClose = () => ({}),
}): JSX.Element => {
  const sections: string[] = ["header", "content", "footer"];

  useEffect(() => {
    document.addEventListener("keydown", handleKey, false);

    return () => {
      document.removeEventListener("keydown", handleKey, false);
    };
  }, []);

  const handleKey = ({ keyCode }: any) =>
    keyCode === ESCAPE_KEY_CODE && closeModal();

  const closeModal = (): void => onClose();
  const handleOverlay = (): void => {
    if (overlay) {
      closeModal();
    }
  };

  const modalClassNames: string = buildClassNames(`${className} modal modal--show`, {
    "modal--alert": isAlert,
    "modal--full-screen": isFullScreen,
    "modal--simple": isSimple,
    "modal--upload": isUpload,
  });

  const [
    modalHeaderClassNames,
    modalContentClassNames,
    modalFooterClassNames,
  ]: string[] = sections.map((alertType: string): string =>
    buildClassNames(`modal__${alertType}`, {
      [`modal__${alertType}--alert`]: isAlert,
      [`modal__${alertType}--simple`]: isSimple,
      [`modal__${alertType}--upload`]: isUpload,
      [`modal__${alertType}--full-screen`]: isUpload,
    })
  );

  return (
    <>
      <div className="overlay" onClick={handleOverlay} />
      <div className={modalClassNames}>
        <div className={modalHeaderClassNames}>
          {!isAlert && <ModalHeader title={title} closeModal={closeModal} />}
        </div>

        <div className={modalContentClassNames}>{children}</div>

        {showFooter && (
          <ModalFooter
            isAlert={isAlert}
            footerClassNames={modalFooterClassNames}
            closeModal={closeModal}
            actions={actions}
          />
        )}
      </div>
    </>
  );
};

Modal.defaultProps = {
  actions: [],
  children: "No content",
  className: "",
  isAlert: false,
  isFullScreen: false,
  isSimple: false,
  isUpload: false,
  overlay: true,
  showFooter: true,
  title: "",
};

export default Modal;
