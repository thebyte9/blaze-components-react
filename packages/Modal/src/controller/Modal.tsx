import { buildClassNames } from '@blaze-react/utils';
import React, { useEffect, FunctionComponent } from 'react';
import { ESCAPE_KEY_CODE } from '../constants';
import { useComponentLogic } from '../hooks/useComponentLogic';
import { DeprecatedModalFooter, ModalFooter } from '../view/ModalFooter';
import { DeprecatedModalHeader, ModalHeader } from '../view/ModalHeader';
import { IModalProps, ModalType } from '../types';
import { ModalView } from '../view/ModalView';

interface DeprecatedIActions {
  textButton: string;
  callback: () => void;
  modifiers?: string[];
}

interface DeprecatedIModalProps {
  title?: string;
  actions: DeprecatedIActions[];
  isSimple?: boolean;
  isUpload?: boolean;
  isFullScreen?: boolean;
  isAlert?: boolean;
  overlay?: boolean;
  children?: any;
  onClose?: () => void;
  showFooter?: boolean;
  className?: string;
}

export const DeprecatedModal: FunctionComponent<DeprecatedIModalProps> = ({
  children,
  isSimple,
  isAlert,
  isFullScreen,
  title,
  isUpload,
  actions,
  overlay,
  className,
  showFooter,
  onClose,
}): JSX.Element => {
  const sections: string[] = ['header', 'content', 'footer'];

  useEffect(() => {
    document.addEventListener('keydown', handleKey, false);

    return () => {
      document.removeEventListener('keydown', handleKey, false);
    };
  }, []);

  const handleKey = ({ keyCode }: any) => keyCode === ESCAPE_KEY_CODE && closeModal();

  const closeModal = (): void => onClose();

  const handleOverlay = (): void => {
    if (overlay) {
      closeModal();
    }
  };

  const modalClassNames: string = buildClassNames(`${className} modal modal--show`, {
    'modal--alert': isAlert,
    'modal--full-screen': isFullScreen,
    'modal--simple': isSimple,
    'modal--upload': isUpload,
  });

  const [modalHeaderClassNames, modalContentClassNames, modalFooterClassNames]: string[] = sections.map(
    (alertType: string): string =>
      buildClassNames(`modal__${alertType}`, {
        [`modal__${alertType}--alert`]: isAlert,
        [`modal__${alertType}--simple`]: isSimple,
        [`modal__${alertType}--upload`]: isUpload,
        [`modal__${alertType}--full-screen`]: isUpload,
      }),
  );

  return (
    <>
      <div className="overlay" onClick={handleOverlay} data-testid="overlay" />
      <div className={modalClassNames} data-testid="modal">
        <div className={modalHeaderClassNames}>
          {!isAlert && <DeprecatedModalHeader title={title} closeModal={closeModal} />}
        </div>

        <div className={modalContentClassNames}>{children}</div>

        {showFooter && (
          <DeprecatedModalFooter
            isAlert={isAlert}
            footerClassNames={modalFooterClassNames}
            closeModal={closeModal}
            actions={actions}
          />
        )}
      </div>
    </>
  );
};

export const Modal = ({
  type = ModalType.OVERLAY,
  title,
  onClose,
  children,
  actions,
  showFooter,
  theme,
  variant = 'default',
  overrides,
  ...rest
}: IModalProps): any => {
  const { container, header, footer } = useComponentLogic({
    componentVariant: variant,
    theme: theme,
    overrides: overrides,
  });

  return (
    <>
      <ModalView classes={container} {...rest}>
        {type !== ModalType.ALERT && <ModalHeader classes={header} title={title} onClose={onClose}></ModalHeader>}
        {children}
        {showFooter && <ModalFooter classes={footer} onClose={onClose} actions={actions} type={type}></ModalFooter>}
      </ModalView>
      <div className="fixed inset-0 z-40 bg-black opacity-25"></div>
    </>
  );
};

DeprecatedModal.defaultProps = {
  actions: [],
  children: 'No content',
  className: '',
  isAlert: false,
  isFullScreen: false,
  isSimple: false,
  isUpload: false,
  overlay: true,
  showFooter: true,
  title: '',
};
