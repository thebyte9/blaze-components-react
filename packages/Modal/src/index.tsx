import withUtils from "@blaze-react/utils";
import React, { useEffect } from "react";
import ModalFooter from "./ModalFooter";
import ModalHeader from "./ModalHeader";

interface IActions {
  textButton: string;
  callback: () => void;
  modifiers?: string[];
}

interface IModalProps {
  title?: string;
  actions: IActions[];
  isSimple: boolean;
  isUpload: boolean;
  isAlert: boolean;
  overlay?: boolean;
  children?: any;
  onClose?: () => void;
  utils: {
    classNames: (className: string | object, classNames?: object) => string;
  };
  className: string;
}
const Modal: React.SFC<IModalProps> = ({
  children,
  isSimple,
  isAlert,
  title,
  isUpload,
  actions,
  overlay,
  utils: { classNames },
  className,
  onClose = () => ({})
}): JSX.Element => {
  const sections: string[] = ["header", "content", "footer"];

  useEffect(() => {
    document.addEventListener("keydown", handleKey, false);

    return () => {
      document.removeEventListener("keydown", handleKey, false);
    };
  }, []);

  const handleKey = ({ keyCode }: any) => keyCode === 27 && closeModal();

  const closeModal = (): void => onClose();

  const modalClassNames: string = classNames(`${className} modal modal--show`, {
    "modal--alert": isAlert,
    "modal--simple": isSimple,
    "modal--upload": isUpload
  });

  const [
    modalHeaderClassNames,
    modalContentClassNames,
    modalFooterClassNames
  ]: string[] = sections.map((alertType: string): string =>
    classNames(`modal__${alertType}`, {
      [`modal__${alertType}--alert`]: isAlert,
      [`modal__${alertType}--simple`]: isSimple,
      [`modal__${alertType}--upload`]: isUpload
    })
  );

  return (
    <>
      {overlay && <div className="overlay" onClick={closeModal} />}
      <div className={modalClassNames} >
        <div className={modalHeaderClassNames}>
          {!isAlert && <ModalHeader title={title} closeModal={closeModal} />}
        </div>

        <div className={modalContentClassNames}>{children}</div>

        <ModalFooter
          isAlert={isAlert}
          footerClassNames={modalFooterClassNames}
          closeModal={closeModal}
          actions={actions}
        />
      </div>
    </>
  );
};
Modal.defaultProps = {
  actions: [],
  children: "No content",
  isAlert: false,
  isSimple: false,
  isUpload: false,
  overlay: true,
  title: ""
};
export default withUtils(Modal);
