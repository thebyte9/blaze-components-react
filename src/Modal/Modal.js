import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

const Modal = ({
  children,
  simple,
  alert,
  title,
  actions,
  isActive,
  buttonText,
  buttonModifiers
}) => {
  const [modalStatus, setModalStatus] = useState(isActive);

  const type = () => {
    if (simple) return '--simple';
    if (alert) return '--alert';
    return '';
  };

  const renderModal = (
    <Fragment>
      <div className="overlay" />
      <div className={`modal modal${type()} modal--show`}>
        <div className={`modal__header modal__header${type()}`}>
          {!alert && <div className="modal__title">{title}</div>}
          {!alert && (
          <div className="modal__close" role="button" onClick={() => setModalStatus(false)}>
            <i className="material-icons">close</i>
          </div>
          )}
        </div>
        <div className={`modal__content modal__content${type()}`}>
          {children}
        </div>
        <div className={`modal__footer modal__footer${type()}`}>
          <div className="modal__button">
            {alert && <Button modifiers="link" onClick={() => setModalStatus(false)}>Cancel</Button>}
            {actions.map(([text, func, modifiers = 'link']) => (
              <Button key={text} modifiers={modifiers} onClick={func}>{text}</Button>
            ))}
          </div>
        </div>
      </div>
    </Fragment>
  );

  return (
    <Fragment>
      {modalStatus && renderModal}
      {
        buttonText
        && (
        <Button
          modifiers={buttonModifiers}
          onClick={() => setModalStatus(!modalStatus)}>
          {buttonText}
        </Button>
        )
      }
    </Fragment>
  );
};

Modal.propTypes = {
  title: PropTypes.string,
  buttonText: PropTypes.string,
  buttonModifiers: PropTypes.string,
  actions: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.string,
      PropTypes.func
    )
  ),
  simple: PropTypes.bool,
  alert: PropTypes.bool,
  isActive: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Modal.defaultProps = {
  title: '',
  buttonText: '',
  buttonModifiers: 'outline',
  actions: [],
  simple: false,
  alert: false,
  isActive: false,
  children: 'No content'
};

export default Modal;
