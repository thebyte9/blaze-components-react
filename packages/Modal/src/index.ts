import { Modal } from './controller/Modal';
import { ModalView } from './view/ModalView';

export default { Modal, ModalView };
