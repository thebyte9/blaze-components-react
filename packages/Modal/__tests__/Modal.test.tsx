import { mount, shallow } from "enzyme";
import expect from "expect";
import React from "react";
import Modal from "../src";

const defaultProps = (override: object = {}) => ({
  actions: [],
  buttonText: "Simple modal",
  isActive: true,
  onChange: () => void 0,
  simple: true,
  ...override
});

describe("Modal component", () => {
  test("should be defined and renders correctly (snapshot)", () => {
    const wrapper = shallow(<Modal {...defaultProps()} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  test("should toggle modal on click button", () => {
    const wrapper = mount(<Modal {...defaultProps()} />);
    wrapper
      .find("Button")
      .at(0)
      .simulate("click");
    expect(wrapper.find(".modal")).toHaveLength(0);
  });

  test("should render and close alert modal", () => {
    const actions = [["delete", () => ({}), "alert small"]];
    const override = {
      actions,
      alert: true,
      isActive: true,
      simple: false
    };
    const wrapper = mount(<Modal {...defaultProps(override)} />);
    expect(wrapper.find(".modal--alert")).toHaveLength(1);
    wrapper
      .find("Button")
      .at(0)
      .simulate("click");
    expect(wrapper.find(".modal--alert")).toHaveLength(0);
  });

  test("should render and close scrollable modal", () => {
    const actions = [["Action 1", () => ({})], ["Action 2", () => ({})]];
    const override = {
      actions,
      alert: false,
      simple: false,
      title: "Scrollable Modal"
    };
    const wrapper = mount(<Modal {...defaultProps(override)} />);
    expect(wrapper.find(".modal__title").text()).toContain("Scrollable Modal");
    wrapper.find(".modal__close").simulate("click");
    expect(wrapper.find(".modal")).toHaveLength(0);
  });
});
