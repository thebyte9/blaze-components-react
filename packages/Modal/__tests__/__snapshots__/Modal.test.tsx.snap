// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should be defined and renders correctly (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="overlay"
  />
  <div
    class=" modal modal--show modal--simple"
  >
    <div
      class="modal__header modal__header--simple"
    >
      <div
        class="modal__title"
      />
      <div
        class="modal__close"
        role="button"
      >
        <i
          class="material-icons"
        >
          close
        </i>
      </div>
    </div>
    <div
      class="modal__content modal__content--simple"
    >
      No content
    </div>
    <div
      class="modal__footer modal__footer--simple"
    >
      <div
        class="modal__button"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal component should close modal on overlay clicked 1`] = `
<DocumentFragment>
  <div
    class="overlay"
  />
  <div
    class=" modal modal--show modal--simple"
  >
    <div
      class="modal__header modal__header--simple"
    >
      <div
        class="modal__title"
      />
      <div
        class="modal__close"
        role="button"
      >
        <i
          class="material-icons"
        >
          close
        </i>
      </div>
    </div>
    <div
      class="modal__content modal__content--simple"
    >
      No content
    </div>
    <div
      class="modal__footer modal__footer--simple"
    >
      <div
        class="modal__button"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Modal component should render and close alert modal 1`] = `
<DocumentFragment>
  <div
    class="overlay"
  />
  <div
    class=" modal modal--show modal--alert modal--simple"
  >
    <div
      class="modal__header modal__header--alert modal__header--simple"
    />
    <div
      class="modal__content modal__content--alert modal__content--simple"
    >
      No content
    </div>
    <div
      class="modal__footer modal__footer--alert modal__footer--simple"
    >
      <div
        class="modal__button"
      >
        <button
          class="button button--cancel"
          type="button"
        >
          Cancel
        </button>
        <button
          class="button button--alert button--small"
          type="button"
        >
          delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
